### ISSUE TITLE
No weight update due to sign function

### ISSUE THREAD

**Issue by jonkoi** [layog/Accurate-Binary-Convolution-Network#6]
Hi,

I don't think weights get updated because of the tf.sign function cutting off backprop gradients:

`binarized_filters = tf.sign(tiled_filters + expanded_stddev, name="binarized_filters")`

i believe this line is needed:

`with tf.get_default_graph().gradient_override_map({"Sign": "Identity"}):`

**Comment by layog** [comment]
Sorry, for the late reply.

Yes, `shift_parameters` and `beta_values` are currently not trainable due to `tf.sign` function. It's mentioned on the bottom of the [README.md](https://github.com/layog/Accurate-Binary-Convolution-Network/blob/master/README.md#testing).

I didn't get time to fix that one but I guess your solution will work, will try and publish the results.

**Comment by martinkersner** [comment]
@jonkoi You are right, the implementation is not correct. The first two ABC layers behave only as constant feature extractors and all the learning happens in the last two float fully connected layers.

**Comment by RealDataLLC** [comment]
Even with the weight update, it seems to not be learning on any dataset beside MNIST. Did you have success @jonkoi?
